### PR TITLE
webContents: exit tabbed fullscreen when esc key is pressed

### LINKED ISSUE #2517: Can't exit html5 fullscreen by keyboard
In most browsers, users can exit html5 fullscreen by pressing 'ESC' key, however Electron can't. I suggest Electron to also do this by default.

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #2517

### CODE REVIEW

**Comment by hokein** [comment]
:+1:

**Comment by magicae** [comment]
:+1:

**Comment by zcbenz** [comment]
We should probably check it in `HandleKeyboardEvent` instead of `PreHandleKeyboardEvent`, otherwise users won't be able to override this behavior.

**Comment by deepak1556** [comment]
@zcbenz Fixed, thanks!

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -277,7 +277,10 @@ bool WebContents::IsPopupOrPanel(const content::WebContents* source) const {
 void WebContents::HandleKeyboardEvent(
     content::WebContents* source,
     const content::NativeWebKeyboardEvent& event) {
-  if (type_ == BROWSER_WINDOW) {
+  if (event.windowsKeyCode == 27 && is_html_fullscreen()) {
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: Use VKEY_ESCAPE instead of 27.

Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: Fixed.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**